metofficedatahub/multiple_files.py

```diff
@@ -99,3 +108,49 @@ def load_all_files(self) -> xr.Dataset:
         dataset = xr.merge(all_dataset)
 
         return dataset
+
+
+def save_to_zarr(dataset: xr.Dataset, save_dir: str, save_latest: bool = True):
+    """
+    Save dataset to zarr file
+
+    :param dataset: The Xarray Dataset to be save
+    :param save_dir: the directory where data is saved.
+        The zarr file will be saved using the timestamp of the run in isoformat
+    :param save_latest: option to save as 'latest.zarr'
+    """
+
+    logger.info(f"Saving data to zarr file here: {save_dir}")
+
+    # Make two files names
+    # 1. use the date timestamp of the data. Idea is that this will keep the historic
+    # 2. the latest - shows the most recent data without search through historic.
+    filename = pd.to_datetime(dataset.time.values).tz_localize("UTC").isoformat()
+    filename_and_path = f"{save_dir}/{filename}.zarr"
+    filename_and_path_latest = f"{save_dir}/latest.zarr/"
```

Comment:
could move this whole thing into a separate function with a more descriptive name, will probably make it easier to read

Reply:
even better: This could also cover the below `extra step needed if we are saving to AWS.`, as that's also related to generating the filename.

You could have something like

```python
filename_and_path, filename_and_path_latest = generate_output_file_names(
    pd,
    save_dir,
    filename,
    is_aws_s3_fs=fsspec.open(save_dir).fs == s3fs.S3FileSystem()
)
```

Reply:
yea good shout - ive done this and it looks much neater now